modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCScopeValidationTest.java

```diff
@@ -0,0 +1,251 @@
+
+    CookieStore cookieStore = new BasicCookieStore();
+
+    protected HttpClient client;
+    protected List<NameValuePair> consentParameters = new ArrayList<>();
+    OIDCApplication playgroundApp;
+    ServiceProvider serviceProvider;
+    private String claimsToGetConsent;
+    private String scopes;
+
+    private static final String INTERNAL_LOGIN_SCOPE = "internal_login";
+    private static final String SYSTEM_SCOPE = "SYSTEM";
+    private static final String INTERNAL_USER_MGT_UPDATE_SCOPE = "internal_user_mgt_update"; // admin permission.
+    private static final String INVALID_SCOPE = "internal_user_update"; // This scope does not exist.
+
+
+    @BeforeMethod(alwaysRun = true)
+    public void testInit(Method method) throws Exception {
+
+        super.init();
+        String scope = OIDCUtilTest.role;
+        if (method.getName().equalsIgnoreCase("testScopeValidationWithAdminUser")) {
+            scope = "admin";
+        }
+        initUser(scope);
+        createUser(user);
+        userInfo.setUserName(user.getUsername());
+        userInfo.setPassword(user.getPassword());
+
+        playgroundApp = initApplication();
+        serviceProvider = createApplication(new ServiceProvider(), playgroundApp);
+        client = HttpClientBuilder.create().setDefaultCookieStore(cookieStore).build();
+
+
+    }
+
+    @AfterMethod(alwaysRun = true)
+    public void testClear() throws Exception {
+
+        deleteUser(user);
+        deleteApplication(playgroundApp);
+        clear();
+    }
+
+
+
+    @Test(groups = "wso2.is", description = "Test scope validation with a user which has 'internal/everyone' role.")
+    public void testScopeValidationWithNormalUser() throws Exception {
+        testSendAuthenticationRequest(playgroundApp, client);
+        testAuthentication(playgroundApp);
+        Assert.assertEquals(claimsToGetConsent, "0_Email,1_First Name",
+                "Requested claims were not prompted to ask the consent.");
+        Assert.assertNotNull(scopes);
+        Assert.assertEquals(INTERNAL_LOGIN_SCOPE, scopes, "'" + INTERNAL_LOGIN_SCOPE + "' should return " +
+                "with the redirect URL");
+        Assert.assertFalse(scopes.contains(SYSTEM_SCOPE), "'" + SYSTEM_SCOPE + "' scopes should not exist after the validation.");
+        Assert.assertFalse(scopes.contains(INTERNAL_USER_MGT_UPDATE_SCOPE),
+                "'" + INTERNAL_USER_MGT_UPDATE_SCOPE + "' scopes should not exist after the validation. " +
+                        "Since user does not have permission for this scope.");
+        Assert.assertFalse(scopes.contains(INVALID_SCOPE), "'" + INVALID_SCOPE + "' scopes should not exist" +
+                " after the validation. This scope does not exist.");
+
+    }
+
+    @Test(groups = "wso2.is", dependsOnMethods = {"testScopeValidationWithNormalUser"}, description = "Test scope" +
+            " validation with an admin user.")
+    public void testScopeValidationWithAdminUser() throws Exception {
```

Comment:
Can't we use a data provider for this and include different sets of scopes?

Reply:
Fixed with https://github.com/wso2/product-is/pull/13004/commits/a0ee361a76ed7d6c50f7c58215fff138992e1fbc